### PR TITLE
Add github action to do CI checks against macOS

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
As the title says. We already do check macOS on Travis but we're in the process  of converting all our CI to Github Actions to make things easier for all of us.

Currently doesn't cache builds because I couldn't get ccache to work, this can be added in a later PR (note that tests take a lot longer than builds anyway so the caching doesn't save us a ton of time).

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: .github/workflows/pull-request-checks.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (danpoe) [inline .github/workflows/pull-request-checks.yaml]: Looks like a comment from the template that could be removed :cup_with_straw:

File: .github/workflows/pull-request-checks.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+name: pull-request-checks
+on:
+  pull_request:
+    branches: [ develop ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
```
Comment (danpoe) [inline .github/workflows/pull-request-checks.yaml]: :cup_with_straw:

File: .github/workflows/pull-request-checks.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  check-macos-10_15-cmake-clang:
+    runs-on: macos-10.15
+    env:
+      # CCache maximum size shouldn't be too big
+      # because we have a global cache limit of 5G.
+      # Full CProver uses about 100MB of cache,
+      # we'll keep 5 times that around per OS
+      # to account for different versions/different
+      # flags
+      # Bump so CI runs again 2
```
Comment (danpoe) [inline .github/workflows/pull-request-checks.yaml]: What does this comment mean? (line 19)

**Review by danpoe — APPROVED** [review]

**Review by NlightNFotis — APPROVED** [review]

File: .github/workflows/pull-request-checks.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  check-macos-10_15-cmake-clang:
+    runs-on: macos-10.15
+    env:
+      # CCache maximum size shouldn't be too big
+      # because we have a global cache limit of 5G.
+      # Full CProver uses about 100MB of cache,
+      # we'll keep 5 times that around per OS
+      # to account for different versions/different
+      # flags
+      CCACHE_MAXSIZE: 500Mi
```
Comment (chrisr-diffblue) [inline .github/workflows/pull-request-checks.yaml]: I guess as you backed out the ccache attempts, this is not needed any longer?

**Review by chrisr-diffblue — APPROVED** [review]
LGTM

**Pull request merged**